samples/contrib/azure-samples/kfp-azure-databricks/tests/test_dbfsblock_op.py

```diff
@@ -0,0 +1,198 @@
+import unittest
+from pathlib import Path
+import kfp
+from kfp.dsl import PipelineParam
+from databricks import CreateDbfsBlockOp, DeleteDbfsBlockOp
+
+class TestCreateDbfsBlockOp(unittest.TestCase):
+
+    def test_databricks_createdbfsblock_without_k8s_or_block_name(self):
+        def my_pipeline():
+            CreateDbfsBlockOp(
+                name="createdbfsitem",
+                data="cHJpbnQoImhlbGxvLCB3b3JsZCIpCgoK",
+                path="/Users/alejacma@microsoft.com/ScalaExampleNotebook"
+            )
+
+        self.assertRaises(ValueError, lambda: kfp.compiler.Compiler()._compile(my_pipeline))
+
+    def test_databricks_createdbfsblock(self):
+        def my_pipeline():
+            block_name = "createdbfsitem"
+            data = "cHJpbnQoImhlbGxvLCB3b3JsZCIpCgoK"
+            path = "/Users/alejacma@microsoft.com/ScalaExampleNotebook"
+            
+
+            expected_spec = {
+                "path": path,
+                "data": data
+               
+            }
+
+            res = CreateDbfsBlockOp(
+                name="createdbfsitem",
+                block_name=block_name,
+                path=path,
+                data=data
+            )
+            
+            self.assert_res(res, expected_spec)
+
+        kfp.compiler.Compiler()._compile(my_pipeline)
+
+    def test_databricks_createdbfsblock_with_spec(self):
+        def my_pipeline():
+            block_name = "createdbfsitem"
+            spec = {
+                "data": "cHJpbnQoImhlbGxvLCB3b3JsZCIpCgoK",
+                "path": "/Users/alejacma@microsoft.com/ScalaExampleNotebook"          
+            }
+
+            res = CreateDbfsBlockOp(
+                name="createdbfsitem",
+                block_name=block_name,
+                spec=spec
+            )
+
+            self.assert_res(res, spec)
+
+        kfp.compiler.Compiler()._compile(my_pipeline)
+
+    def test_databricks_createdbfsblock_with_spec_and_extra_args(self):
```

Comment:
The idea of this test is to show that you can pass a spec that is missing some field e.g. "data" or has some default value in that field, and you can pass that spec and a new value for the field "data" to the Op and you will get the full spec including the new "data"